packages/flame_tiled/lib/src/renderable_tile_map.dart

```diff
@@ -80,23 +97,31 @@ class RenderableTiledMap {
         result[src] = await SpriteBatch.load(src);
       }
     });
+
     return result;
   }
 
   void _fillBatches() {
-    for (final batch in batches.keys) {
-      batches[batch]!.clear();
-    }
+    batchesByLayer.forEach(
+      (batchMap) => batchMap.values.forEach((batch) => batch.clear),
+    );
 
-    map.layers
-        .where((layer) => layer.visible)
-        .whereType<TileLayer>()
+    var layerInd = 0;
+    _renderableTileLayers(map)
         .map((e) => e.tileData)
         .whereType<List<List<Gid>>>()
-        .forEach(_renderLayer);
+        .forEach(
```

Comment:
You could use `forEachIndexed` here from `'package:collection/collection.dart'`, that collection package is in flutter, but you have to add it to the pubspec file if it isn't there already:
```yaml
  collection: ^1.15.0
  ```

Reply:
Wasn't sure if it'd be ok to add another dependency for that, but it's much cleaner with it. Thanks!

Reply:
Yeah, normally not, but this collection package is already in Flame so we can just as well have it here too.